### ISSUE TITLE
Disable cookies/history/... per-container

### ISSUE THREAD

**Issue by nt1m** [mozilla/testpilot-containers#47]
And possibly consider private-browsing as a container.

**Comment by bakulf** [comment]
What do you mean with this bug?
Cookies are separate per container and you can see them, in about:preferences. For each cookie you can see which container it belongs to.

**Comment by nt1m** [comment]
@bakulf Sorry, I was unclear about this, I've split this out in 2 issues: this one and #48.

#48 describes the use-case for the cookie part. As for disabling history, sometimes I just need a container that disables history without necessarily logging me out of every website. Private browsing provides the latter, but not the former.

With both this issue and #48 combined, we can possibly consider making private browsing a container.

**Comment by TanviHacks** [comment]
@nt1m To be clear, you want a mode that doesn't save history but does save cookies so that you aren't logged out of sites?

In private mode, history is not saved and cookies are wiped when you close private browsing.

In Containers, history is saved and cookies are saved in that container.  Deleting a container will delete the cookies and other storage associated with it, but it won't clear the history entries, as history isn't tagged by Container ID yet.  (History isn't tagged because tagging history is complicated and we haven't figured out if we need to introduce that complexity into the platform for Containers to be successful.)

**Comment by nt1m** [comment]
@TanviHacks Yes, and I do think a setting for each container would do that perfectly. Potentially in the future, there will be no line between private browsing and containers.

**Comment by gerdneuman** [comment]
> History isn't tagged because tagging history is complicated and we haven't figured out if we need to introduce that complexity into the platform for Containers to be successful.

To give some feedback: This issue is the main reason why I stopped using Container tabs. My use case is doing some amazon, twitter, private mail while being at work. I guess everybody does from time to time. So I have my "personal" container. 

Of course I do not use this "personal" container when a colleague (or boss) sits next to me while we're working together on a problem. Also I do not use it when doing some presentation or similar. Still, the URLs that I used (very often) show up in the Awesomebar (URL/location bar) whenever I am in a not-container context. So I type "a" and "amazon" shows up, I type "web" and "webmail.myprovider.com" shows up, etc.

I think associating history to a container would solve this. Using private browsing also does not suffice my use case because then I have absolutely no history at all 

(in the end it's also the same for users regularly watching porn, they might want to create a 'personal' container where they have some hidden history which cannot accidentically show up instead of using private browsing without any history whatsoever)

**Comment by ArchangeGabriel** [comment]
My use case is quite similar: I want my history to be kept for some contexts (like when I navigate in my FLOSS container), but not in other (personal container, shopping one, whatever). Main difference is just that I don’t want my history to be cluttered by thing I don’t need.

**Comment by TuxyKey** [comment]
Hi there,
First of i wanted to thank you guys for an awesome plugin, great work.
My request is in line with the History tagging / grouping / isolating what ever you want to call it.
i basically want to be able to delete / clean the history for one container if possible.
simple example, signing containers stay relatively clean but my shopping containers does not. So cleaning up all the mess there would be helpful. Or my twitter container were i read a lot and follow links, that one needs more cleaning but i would rather not login to all my separate emails every time i do a clean up.
Thanks guys appreciate it. this plugin is the best one i have seen in a while for my needs. it's up there with ublock-origin....
i'm on OSX 10.9.5 . Firefox 55.03

(hope this is the correct place to put my suggestion / request i didn't want to add a new issue as this one is related.

**Comment by AdrienPoupa** [comment]
+1 for this, a heavy use of different containers (eg one for Facebook, one for Github, one for Google) makes navigation a pain.

**Comment by Techwolfy** [comment]
@bakulf @TanviHacks What's the status of this bug? The parallel bug on Bugzilla (https://bugzilla.mozilla.org/show_bug.cgi?id=1353726) hasn't seen any activity since last April.

I've been stuck on Firefox 56 for months as it's the latest browser that supports this feature (via https://github.com/Infocatcher/Private_Tab), and I've just gotten a [Heartbeat](https://wiki.mozilla.org/Firefox/Shield/Heartbeat) popup warning me that it's now critically out of date, which is true. I'd like to try out Quantum but it's not worth breaking my workflow, and I can't downgrade to ESR due to breaking changes in the places database from FF 55.

**Comment by groovecoder** [comment]
To be frank & honest, you shouldn't wait for this feature to update Firefox. Some exploratory work has started on assigning settings (like `browser.*history*`) values to specific containers, but it will need many months of development time, let alone testing & release time.

In the meantime, you're are running Firefox vulnerable to a number of security threats, including Specter and Meltdown.

**Comment by stoically** [comment]
Sorry to hijack this thread with a shameless plug - but since I think some are interested in it, I'll just mention it here: The Temporary Containers Add-on now as a feature to automatically delete History of Temporary Containers. Since this isn't fully supported by Firefox yet, it will delete URLs you visit in "Deletes History Temporary Containers" completely from your history - that's why the feature comes with a big red warning before you're allowed to use it in an Opt-in-fashion.

**Comment by MurzNN** [comment]
Here is Firefox issue about store container id in history database: [Make History aware of userContextId](https://bugzilla.mozilla.org/show_bug.cgi?id=1283320)

**Comment by Nivatius** [comment]
Cookie Quick Manager Makes this possible https://addons.mozilla.org/de/firefox/addon/cookie-quick-manager/

**Comment by TuxyKey** [comment]
> Cookie Quick Manager Makes this possible https://addons.mozilla.org/de/firefox/addon/cookie-quick-manager/

Hi Nivatius,
Thanks for the tip/help i tried out cookie-quick-manager. and it's awesome.
works really great with "Multi-Account_containers".
For others reading this here's a short summary how to quickly use this add-on.
To delete all cookies from a specific account container i discovered the fasted way to do so is to open a specific account container klick on the cookie manager icon and select "Delete current context cookies". to double check say the first time so you know it works .. go in to settings and select the correct context in the drop down menu in the middle of the settings page.

Really happy with this add-on. Really makes multi-account-container even more powerful and useful to  me

**Comment by xieve** [comment]
I hacked together an experimental extension that tries to do this with the existing APIs. It does actually delete just the visits, not the entire history entries like Temporary Containers. It's not a very pretty implementation though, nor is it particularly efficient. Feel free to try, maybe I'll release it on the Firefox Add-On page sometime.

**Comment by anewuser** [comment]
This new lightweight extension can open temporary containers too: [Open in Temp Container](https://addons.mozilla.org/addon/open-in-temp-container/). Give it access to your browsing history for it to prevent new URLs from being stored in your history.
